crates/core/src/network/handler/mod.rs

```diff
@@ -483,27 +483,49 @@ fn get_meta_aspects_from_dht_eav(
             let header = value_entry.headers[0].to_owned();
 
             match eavi.attribute() {
-                Attribute::LinkTag(_, _) => {
-                    let link_data = unwrap_to!(value_entry.entry_with_meta.entry => Entry::LinkAdd);
-                    Ok(EntryAspect::LinkAdd(link_data.clone(), header))
-                }
-                Attribute::RemovedLink(_, _) => {
-                    let (link_data, removed_link_entries) =
-                        unwrap_to!(value_entry.entry_with_meta.entry => Entry::LinkRemove);
-                    Ok(EntryAspect::LinkRemove(
-                        (link_data.clone(), removed_link_entries.clone()),
-                        header,
-                    ))
+                Attribute::LinkTag(_, _) => match value_entry.entry_with_meta.entry {
+                    Entry::LinkAdd(link_data) | Entry::LinkRemove((link_data, _)) => {
+                        Ok(EntryAspect::LinkAdd(link_data, header))
+                    }
+                    _ => Err(HolochainError::from("Invalid Entry Value for Link Add")),
+                },
+                Attribute::RemovedLink(link_type, link_tag) => {
+                    match value_entry.entry_with_meta.entry {
+                        Entry::LinkRemove((link_data, removed_link_entries)) => Ok(
+                            EntryAspect::LinkRemove((link_data, removed_link_entries), header),
+                        ),
+                        Entry::LinkAdd(link_data) => {
+                            let links = get_links(
+                                &context,
+                                entry_address.clone(),
+                                Some(link_type),
+                                Some(link_tag),
+                                Some(CrudStatus::Live),
+                                GetLinksQueryConfiguration::default(),
+                            )?;
+
+                            //not very efficient but nice to fall back on
+                            let filtered_links = links
+                                .into_iter()
+                                .filter(|link_for_filter| {
+                                    &link_for_filter.target == link_data.link().target()
+                                })
+                                .map(|s| s.address)
+                                .collect::<Vec<_>>();
+
+                            Ok(EntryAspect::LinkRemove((link_data, filtered_links), header))
```

Comment:
There is so much going on here, I know it's logic to deal with tombstones, but it's so unintuitive that a Entry::LinkAdd would lead to a EntryAspect::LinkRemove, can you add some comments as to what's going on here?